### PR TITLE
Prevent AILTN publisher background task from ending prematurely.

### PR DESCRIPTION
**Pull request by brandonpage**

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m [hunk 1]
```diff
@@ -243,6 +245,7 @@ - (void) publishEvents:(NSArray<SFSDKInstrumentationEvent *> *) events {
                 }
                 publishCompleteBlock = nil;
             }
+            [self cleanupBackgroundTask];
```
Comment (trooper2013) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: @brandonpage Shouldn't this  clean up occur in the else block above it?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m [hunk 2]
```diff
@@ -313,14 +316,10 @@ - (void) publishOnAppBackground {
         return;
     }
     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        __block UIBackgroundTaskIdentifier task;
         task = [[SFApplicationHelper sharedApplication] beginBackgroundTaskWithName:NSStringFromClass([self class]) expirationHandler:^{
-            [[SFApplicationHelper sharedApplication] endBackgroundTask:task];
-            task = UIBackgroundTaskInvalid;
+            [self cleanupBackgroundTask];
```
Comment (trooper2013) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: Is the retain intentional?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m (hunk 1 above)
Comment (brandonpage) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: Don't we want to end the background task regardless of success/failure? If the operation fails the events aren't deleted and will be sent (with new ones) the next time the app is backgrounded.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m (hunk 2 above)
Comment (trooper2013) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Analytics/SFSDKSalesforceAnalyticsManager.m]: @brandonpage on a second thought this is ok 👍

**Pull request merged**